DOCS OBSDOCS-904 - Logging 5.8.5 Release Notes (#73271)
openshift/openshift-docs: modules/logging-release-notes-5-8-5.adoc

// module included in /logging/logging-5-8-release-notes
:_mod-docs-content-type: REFERENCE
[id="logging-release-notes-5-8-5_{context}"]
= Logging 5.8.5
This release includes link:https://access.redhat.com/errata/RHBA-2024:1474[OpenShift Logging Bug Fix Release 5.8.5].

[id="logging-release-notes-5-8-5-bug-fixes"]
== Bug fixes

> * Before this update, the configuration of the `ServiceMonitor` by Loki Operator could match many Kubernetes services, which led to Loki Operator's metrics being collected multiple times. With this update, the `ServiceMonitor` setup now only matches the dedicated metrics service. (link:https://issues.redhat.com/browse/LOG-5250[LOG-5250])

Reviewer: ```suggestion
* Before this update, the configuration of the Loki Operator's `ServiceMonitor` could match many Kubernetes services, resulting in the Loki Operator's metrics being collected multiple times. With this update, the configuration of `ServiceMonitor` now only matches the dedicated metrics service. (link:https://issues.redhat.com/browse/LOG-5250[LOG-5250])
```

> * Before this update, the Red Hat build pipeline didn't use the existing build details in Loki builds, leaving out information like revision, branch, and version. With this update, the Red Hat build pipeline now adds these inputs to the Loki builds, fixing the issue. (link:https://issues.redhat.com/browse/LOG-5201[LOG-5201])

Reviewer: ```suggestion
* Before this update, the Red Hat build pipeline didn't use the existing build details in Loki builds and omitted information such as revision, branch, and version. With this update, the Red Hat build pipeline now adds these details to the Loki builds, fixing the issue. (link:https://issues.redhat.com/browse/LOG-5201[LOG-5201])
```

> * Before this update, the Loki Operator checked whether the pods were running to decide if the `LokiStack` was ready. With this update, it also checks if the pods are ready, so the readiness of the `LokiStack` reflects the state of its components. (link:https://issues.redhat.com/browse/LOG-5171[LOG-5171])

Reviewer: ```suggestion
* Before this update, the Loki Operator checked if the pods were running to decide if the `LokiStack` was ready. With this update, it also checks if the pods are ready, so that the readiness of the `LokiStack` reflects the state of its components. (link:https://issues.redhat.com/browse/LOG-5171[LOG-5171])
```

> * Before this update, executing a query for logs metrics caused an error on the histogram. With this update, the histogram toggle and chart are disabled and hidden because the histogram doesn't work with log metrics. (link:https://issues.redhat.com/browse/LOG-5044[LOG-5044])

Reviewer: ```suggestion
* Before this update, running a query for log metrics caused an error in the histogram. With this update, the histogram toggle function and the chart are disabled and hidden because the histogram doesn't work with log metrics. (link:https://issues.redhat.com/browse/LOG-5044[LOG-5044])
```

> * Before this update, the Loki and Elasticsearch bundle had the wrong `maxOpenShiftVersion`, triggering `IncompatibleOperatorsInstalled` alerts. With this update, including 4.16 as the `maxOpenShiftVersion` property in the bundle fixes the issue. (link:https://issues.redhat.com/browse/LOG-5272[LOG-5272])

Reviewer: ```suggestion
* Before this update, the Loki and Elasticsearch bundle had the wrong `maxOpenShiftVersion`, resulting in `IncompatibleOperatorsInstalled` alerts. This update adds the `4.16` property to the bundle as `maxOpenShiftVersion`, which fixes the issue. (link:https://issues.redhat.com/browse/LOG-5272[LOG-5272])
```

Author: Ignoring this comment as it isn't as per release notes guidelines. Only changing`triggering` to `resulting in`.

> * Before this update, the build pipeline did not include linker flags for the build date, causing Loki builds to show empty strings for `buildDate` and `goVersion`. With this update, adding the missing linker flags in the build pipeline fixes the issue. (link:https://issues.redhat.com/browse/LOG-5274[LOG-5274])

Reviewer: ```suggestion
* Before this update, the build pipeline did not contain linker flags for the build date, so Loki builds displayed empty strings for `buildDate` and `goVersion`. This update adds the missing linker flags to the build pipeline, which fixes the issue. (link:https://issues.redhat.com/browse/LOG-5274[LOG-5274])
```

Author: Same as above. Suggestion not aligning with OpenShift release notes guidelines.

> * Before this update, a bug in LogQL parsing left out some line filters from the query. With this update, the parsing now includes all the line filters while keeping the original query unchanged. (link:https://issues.redhat.com/browse/LOG-5270[LOG-5270])

Reviewer: ```suggestion
* Before this update, an error in LogQL parsing caused some row filters to be omitted from the query. With this update, all row filters are now taken into account, while the original query remains unchanged. (link:https://issues.redhat.com/browse/LOG-5270[LOG-5270])
```

Author: I'm ignoring the passive voice suggestion. Also, the second line suggestion is missing a definite article.

> * Before this update, the Loki operator ServiceMonitor in openshift-operators-redhat used static token and CA files for authentication, causing errors in the Prometheus operator in User-Workload-Monitoring on such ServiceMonitor configuration. With this update, the Loki operator ServiceMonitor in openshift-operators-redhat now references a serviceaccount token secret via a LocalReference. This approach resolves the issue, allowing the User-Workload-Monitoring Prometheus operator to handle the Loki operator ServiceMonitor successfully, enabling Prometheus to scrape the Loki operator metrics. (link:https://issues.redhat.com/browse/LOG-5240[LOG-5240])

Reviewer: ```suggestion
* Before this update, the Loki Operator `ServiceMonitor` in `openshift-operators-redhat` used static token and CA files for authentication, which caused errors in the Prometheus Operator in user workload monitoring with such a `ServiceMonitor` configuration. With this update, the Loki Operator `ServiceMonitor` in `openshift-operators-redhat` now references a serviceaccount token secret through a `LocalReference`. This approach fixes the issue and enables the Prometheus Operator user workload monitoring to successfully handle the Loki Operator `ServiceMonitor` so that Prometheus can retrieve the Loki Operator metrics. (link:https://issues.redhat.com/browse/LOG-5240[LOG-5240])
```
